### PR TITLE
feat: Add PR Dashboard

### LINKED ISSUE #499: Add a PR Dashboard
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Currently there is not an easy way for contributors to see clearly if their PR meets the requirements for proper commits. This creates additional work for the maintainer team and prolongs the review process. 

### 🛠️ Implementation Notes

Implement a PR Dashboard similar to what is shown in the Hiero-SDK-C++ Repository. 

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### PR DESCRIPTION
**Pull request by aceppaluni**
# Description

This PR adds a GitHub-action to present contributors with a PR dashboard that allows the to see the status of their commits. This dashboard updates on commit pushes ensuring that each PR meets the proper requirements for allowing commit signing and merging. 

## Changes Made
- [x] Added...
* A script folder containing the bot use scripts executed by the corresponding YML files.
* Helper methods were added to assist bot logic scripts and consolidate infrastructure.  
- [ ] Modified...
- [ ] Fixed...

## Related Issues

Fixes: #499 

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [x] Branch up-to-date with main

## Deployment Notes

Links to testing: 
* https://github.com/aceppaluni/hiero-website/pull/14
* https://github.com/aceppaluni/hiero-website/pull/12

## Summary by CodeRabbit

* **New Features**
  * Introduced automated PR validation system that checks commit signatures, merge conflicts, and issue linkage
  * Auto-assigns pull request authors and dynamically manages PR status labels based on validation results
  * Posts automated summary comments on PRs with comprehensive check results
  * Added configurable automation settings for teams, labels, and skill hierarchies to support contributor workflows

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "nextjs",
   "version": "0.1.0",
-  "type": "module",
+  "type": "commonjs",
```
Comment (danielmarv) [inline package.json]:
```suggestion
  "type": "module",
```
this has to be reverted to module coz the whole importation architecture is in the module format
and was it intended coz i se it is out of scope

**Review by danielmarv — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (danielmarv) [reply package.json]: And it is the cause of all failing workflows

Comment (aceppaluni) [reply package.json]: Yes am fixing this now

File: .github/scripts/helpers/api.js [hunk 2]
```diff
@@ -0,0 +1,1144 @@
...
+} from './constants.js';
+
+import {
+  checkDCO,
+  checkGPG,
+  checkMergeConflict,
+  checkIssueLink,
+} from './checks.js';
+
+import { buildBotComment } from './comments.js';
```
[the remark this answers is not in the document]
Comment (danielmarv) [reply .github/scripts/helpers/api.js:27]: what do you think about this suggestion?

**Comment by aceppaluni** [comment]
@danielmarv 
Am addressing these changes here https://github.com/hiero-ledger/hiero-enterprise-java/pull/223

Once all is addressed, I will update these files and close the remaining conversations. 

Thank you!! :)

File: .github/workflows/on-pr-update.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      - edited
+
+permissions:
+  contents: read
+  pull-requests: write
+  checks: write
+
+jobs:
+  on-pr-update:
+    runs-on: hiero-client-sdk-linux-large
```
Comment (andrewb1269) [inline .github/workflows/on-pr-update.yml]:
```suggestion
    runs-on: hl-web-lin-md
```

File: .github/workflows/on-pr.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      - ready_for_review
+
+permissions:
+  pull-requests: write
+  contents: read
+  checks: write
+
+jobs:
+  on-pr-open:
+    runs-on: ubuntu-latest
```
Comment (andrewb1269) [inline .github/workflows/on-pr.yml]:
```suggestion
    runs-on: hl-web-lin-md
```

**Review by andrewb1269 — DISMISSED** [review]

**Comment by aceppaluni** [comment]
@danielmarv This is now ready to be reviewed.

**Review by danielmarv — APPROVED** [review]
